docs/data/material/customization/theme-components/theme-components.md

```diff
@@ -48,9 +48,22 @@ const theme = createTheme({
 
 {{"demo": "GlobalThemeOverride.js"}}
 
-The list of each component's classes is documented under the **CSS** section of its API page.
+Note that each component is composed of several different parts. If you want to know how to target a specific part of a given component, check the **CSS** section of its API page to see every class it has available. You should then use it inside the `stylesOverrides` key.
 
-To override a lab component's styles with TypeScript, check [this section of the documentation](/components/about-the-lab/#typescript).
+```js
+const theme = createTheme({
+  components: {
+    MuiButton: {
+      styleOverrides: {
+        containedPrimary: {
+          backgroundColor: '#202020',
+          color: '#fff',
+        },
+      },
+    },
+  },
+});
```

Comment:
```suggestion
const theme = createTheme({
  components: {
    MuiButton: {
      styleOverrides: ({ ownerState }) => ({
        ...ownerState.variant === 'contained' && ownerState.color === 'primary' && {
          backgroundColor: '#202020',
          color: '#fff',
        },
      }),
    },
  },
});
```

Reply:
I don't really have all developer context to judge whether this is better or not but I do need to confess that it just looks a bit uglier 😅